src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -34,64 +34,78 @@
 import ai.rapids.cudf.Table;
 
 public class GpuTimeZoneDB {
-
-  public static final int TIMEOUT_SECS = 300;
-
+  private boolean cacheEmpty = true;
 
   // For the timezone database, we store the transitions in a ColumnVector that is a list of 
   // structs. The type of this column vector is:
   //   LIST<STRUCT<utcInstant: int64, localInstant: int64, offset: int32>>
-  private CompletableFuture<Map<String, Integer>> zoneIdToTableFuture;
-  private CompletableFuture<HostColumnVector> fixedTransitionsFuture;
+  private Map<String, Integer> zoneIdToTable;
+  private HostColumnVector fixedTransitions;
 
-  private boolean closed = false;
-
-  GpuTimeZoneDB() {
-    zoneIdToTableFuture = new CompletableFuture<>();
-    fixedTransitionsFuture = new CompletableFuture<>();
+  private GpuTimeZoneDB() {
   }
   
-  private static GpuTimeZoneDB instance = new GpuTimeZoneDB();
+  private static final GpuTimeZoneDB instance = new GpuTimeZoneDB();
   // This method is default visibility for testing purposes only. The instance will be never be exposed publicly
   // for this class.
   static GpuTimeZoneDB getInstance() {
     return instance;
   }
-  
+
+  /**
+   * This should be called on startup of an executor.
+   */
+  public static void cacheDatabaseAsync() {
+    Runnable runnable = () -> {
+      try {
+        cacheDatabase();
+      } catch (Exception e) {
+        // ignore
+      }
+    };
+    Thread thread = Executors.defaultThreadFactory().newThread(runnable);
+    thread.setName("gpu-timezone-database-0");
+    thread.setDaemon(true);
+    thread.start();
+  }
+
   /**
-   * Start to cache the database. This should be called on startup of an executor. It should start
-   * to cache the data on the CPU in a background thread. It should return immediately and allow the
-   * other APIs to be called. Depending on what we want to do we can have the other APIs block
-   * until this is done caching, or we can have private APIs that would let us load and use specific
-   * parts of the database. I prefer the former solution at least until we see a performance hit
-   * where we are waiting on the database to finish loading.
+   * cache the database.
    */
   public static void cacheDatabase() {
-    synchronized (instance) {
-      if (!instance.isLoaded()) {
-        Executor executor = Executors.newSingleThreadExecutor(
-          new ThreadFactory() {
-            private ThreadFactory defaultFactory = Executors.defaultThreadFactory();
+    instance.cacheDatabaseImpl();
+  }
 
-            @Override
-            public Thread newThread(Runnable r) {
-              Thread thread = defaultFactory.newThread(r);
-              thread.setName("gpu-timezone-database-0");
-              thread.setDaemon(true);
-              return thread;
-            }
-          });
-        instance.loadData(executor);
+  public static void shutdown() {
+    instance.shutdownImpl();
+  }
+
+  private synchronized void cacheDatabaseImpl() {
+    if (cacheEmpty) {
```

Comment:
Same race as described on shutdown.  What if async load on startup has not completed yet before a method is called that blindly calls cacheDatabase?  Seems like we could have to separate threads trying to load the database at that point (caller thread and async thread).  I think we need to check if there's a pending async load in-flight and wait on the thread if that's the case.

Reply:
Ah, synchronized method solves the issue here, but the race on shutdown is still there.  shutdown could "win" the race to the lock followed by async thread, leaving a cached database in memory despite being shutdown.

Reply:
The caches APIs are synchronized, so multiple callings will be serial and wait other to be done.
```scala
private synchronized void cacheDatabaseImpl() {
private synchronized void shutdownImpl() {
```